lib/codecs/src/decoding/format/influxdb.rs

```diff
@@ -0,0 +1,217 @@
+}
+
+impl InfluxdbDeserializer {
+    /// new constructs a new InfluxdbDeserializer
+    pub fn new(lossy: bool) -> Self {
+        Self { lossy }
+    }
+}
+
+impl Deserializer for InfluxdbDeserializer {
+    fn parse(
+        &self,
+        bytes: Bytes,
+        _log_namespace: LogNamespace,
+    ) -> vector_common::Result<SmallVec<[Event; 1]>> {
+        let line: Cow<str> = match self.lossy {
+            true => String::from_utf8_lossy(&bytes),
+            false => Cow::from(std::str::from_utf8(&bytes)?),
+        };
+        let parsed_line = influxdb_line_protocol::parse_lines(&line);
+
+        let res = parsed_line
+            .collect::<Result<Vec<_>, _>>()?
+            .iter()
+            .flat_map(|line| {
+                let ParsedLine {
+                    series,
+                    field_set,
+                    timestamp,
+                } = line;
+
+                field_set
+                    .iter()
+                    .filter_map(|f| {
+                        let measurement = series.measurement.clone();
+                        let tags = series.tag_set.as_ref().clone();
+                        let val = match f.1 {
+                            FieldValue::I64(v) => v as f64,
+                            FieldValue::U64(v) => v as f64,
+                            FieldValue::F64(v) => v,
+                            FieldValue::Boolean(v) => {
+                                if v {
+                                    1.0
+                                } else {
+                                    0.0
+                                }
+                            }
+                            // String values cannot be modelled in our schema
+                            FieldValue::String(_) => return None,
+                        };
+                        Some(Event::Metric(
+                            Metric::new(
+                                format!("{0}_{1}", measurement, f.0),
+                                MetricKind::Absolute,
+                                MetricValue::Gauge { value: val },
+                            )
+                            .with_tags(tags.and_then(|ts| {
+                                Some(MetricTags::from_iter(
+                                    ts.iter().map(|t| (t.0.to_string(), t.1.to_string())),
+                                ))
+                            }))
+                            .with_timestamp(timestamp.and_then(|t| {
+                                NaiveDateTime::from_timestamp_micros(t).map(|t| t.and_utc())
+                            })),
+                        ))
+                    })
+                    .collect::<Vec<_>>()
+            })
+            .collect();
+
+        Ok(res)
+    }
+}
+
+impl From<&InfluxdbDeserializerConfig> for InfluxdbDeserializer {
+    fn from(config: &InfluxdbDeserializerConfig) -> Self {
+        Self {
+            lossy: config.influxdb.lossy,
+        }
+    }
+}
+
+#[cfg(test)]
```

Comment:
Can we add a test per metric type?

Reply:
Based on this [comment](https://github.com/vectordotdev/vector/issues/11337#issuecomment-1897558600), it should be okay to use the gauge data type for Influx's 'measurement,' right? Since we don't have encoding for InfluxDB output.

Reply:
Should we treat other metric types as an error?

Reply:
@pront Sorry to get back to you. Do we need that? The format for Influx line protocol is like this and doesn't specify any metric types: `cpu,host=A,region=west usage_system=64i,usage_user=10i 1590488773254420000`. We're just parsing it into the gauge metric type. Since the input is always like this, does the metric type matter? I fixed the formatting issue. In the upcoming PR, I can add support for InfluxDB encoding so we can specify the metric type tag and support all metric types. Is it okay to add encoding for InfluxDB line protocol?